sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -468,12 +477,26 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
       s"Unsupported value type ${v.getClass.getName} ($v).")
   }
 
+  private def toAttributes(cols: Seq[String]): Seq[Attribute] = {
+    def resolve(colName: String) : Attribute = {
+      df.col(colName).named.toAttribute match {
+        case a: Attribute => a
+        case _ => throw new IllegalArgumentException(s"'$colName' is not a top level column.")
```

Comment:
Can we add a test for this branch? The code seems not to work as expected.  `toAttribute` always return an Attribute and we never hit this branch.

Reply:
(Let me merge different threads here)

The previous implementation resolved the column names as follow:
1) Check the column names against the schema (`df.schema.fields`)
2) Only the column names that matched the top level fields were used for `df.col`.

So, if we want to keep the previous behavior of `fill` (for handling `*`, and etc.), we first need to check the column names against the schema, then do `df.col(colName).named.toAttribute`. 

Then, the nested field will not pass the schema check (since it's checking against the top-level fields), and `toAttribute` will always return `Attribute`. So, I am not sure how this branch can be tested. What do you think?

Reply:
Yea we should call `df.col` to handle `*` etc. But we shouldn't call `.named.toAttribute` which turns everything to attribute and make us not able to detect nested fields.

Can we add a test to fill nested fields and see the result?

Reply:
The nested fields are not supported since the name is checked first against `df.schema.fields` which contains the top-level field, and these do not become candidates for `fill`. `*` is handled the same way and it is ignored. I can capture this behavior in the unit test.

Also, what's the best way to convert `Column` to `Attribute`? Now that we have a single point of entry `def fillValue[T](value: T, cols: Seq[Attribute])`, we need to convert `Column` (result of `df.col`) to `Attribute`.

Reply:
Yes nested fields should be ignored, we should have a test to verify it. how about
```
cols.map(name => df.col(name).expr).collect {
  case a: Attribute => a
}
```

Reply:
great, thanks!